server/channels/api4/bot_test.go

```diff
@@ -1214,7 +1248,8 @@ func TestEnableBot(t *testing.T) {
 	t.Run("enable bot without permission, but with read permission", func(t *testing.T) {
 		th := Setup(t).InitBasic()
 		defer th.TearDown()
-		defer th.RestoreDefaultRolePermissions(th.SaveDefaultRolePermissions())
+		defaultPerms := th.SaveDefaultRolePermissions()
+		defer th.RestoreDefaultRolePermissions(defaultPerms)
```

Comment:
The method name `SaveDefaultRolePermissions` is misleading. The method only returns the current permission.

Anyway, I wonder if we can simply the coder by adding a new method called `RestoreDefaultRolePermissions` that uses `model.MakeDefaultRoles`. Then, the code becomes

```golang
defer th.RestoreDefaultRolePermissions()
```

Reply:
I took a look. `model.MakeDefaultRoles`  returns a lot more roles than `SaveDefaultRolePermissions`. I don't think that would be a 1:1 match. I agree with you that the name is misleading. Maybe `GetDefaultRolePermissions` might be a better name?

Reply:
Can we pick the roles we want from `model.MakeDefaultRoles` using the list that is already in `SaveDefaultRolePermissions`?

Reply:
I guess so, let me see what I can do.

Reply:
I understand the reason for this now. In `th.RestoreDefaultRolePermissions`, there is a block:

```
if strings.Join(role.Permissions, " ") == strings.Join(permissions, " ") {
      continue
}
```

This is needed because `th.UpdateRole` is not an upsert and will fail if the same role name already exists. Only if we pass an existing role with an id, does it skip the role creation part. Therefore, `model.MakeDefaultRoles` won't work because it doesn't have a role ID because it's not saved in the DB yet.

Therefore, I think we have to live with this approach for the time being.